gobblin-data-management/src/main/java/org/apache/gobblin/data/management/conversion/hive/materializer/HiveMaterializer.java

```diff
@@ -0,0 +1,155 @@
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gobblin.data.management.conversion.hive.materializer;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.apache.gobblin.configuration.WorkUnitState;
+import org.apache.gobblin.data.management.conversion.hive.entities.QueryBasedHivePublishEntity;
+import org.apache.gobblin.data.management.conversion.hive.entities.StageableTableMetadata;
+import org.apache.gobblin.data.management.conversion.hive.source.HiveSource;
+import org.apache.gobblin.data.management.conversion.hive.source.HiveWorkUnit;
+import org.apache.gobblin.data.management.conversion.hive.task.HiveConverterUtils;
+import org.apache.gobblin.data.management.conversion.hive.task.HiveTask;
+import org.apache.gobblin.data.management.conversion.hive.task.QueryGenerator;
+import org.apache.gobblin.data.management.copy.hive.HiveDataset;
+import org.apache.gobblin.runtime.TaskContext;
+import org.apache.gobblin.runtime.task.TaskUtils;
+import org.apache.gobblin.source.workunit.WorkUnit;
+
+import com.google.common.base.Strings;
+
+import javax.annotation.Nullable;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+
+/**
+ * A simple {@link HiveTask} for Hive view materialization.
+ */
+public class HiveMaterializer extends HiveTask {
+
+  protected static final String STAGEABLE_TABLE_METADATA_KEY = "internal.hiveMaterializer.stageableTableMetadata";
+  protected static final String MATERIALIZER_MODE_KEY = "internal.hiveMaterializer.materializerMode";
+  protected static final String STORAGE_FORMAT_KEY = "internal.hiveMaterializer.storageFormat";
+  protected static final String QUERY_TO_MATERIALIZE_KEY = "internal.hiveMaterializer.queryToMaterialize";
+
+  /**
+   * Create a work unit to copy a source table to a target table using a staging table in between.
+   * @param dataset {@link HiveDataset} for the source table.
+   * @param destinationTable {@link StageableTableMetadata} specifying staging and target tables metadata.
+   */
+  public static HiveWorkUnit tableCopyWorkUnit(HiveDataset dataset, StageableTableMetadata destinationTable,
+      @Nullable String partitionName) {
+    HiveWorkUnit workUnit = new HiveWorkUnit(dataset);
+    workUnit.setProp(MATERIALIZER_MODE_KEY, MaterializerMode.TABLE_COPY.name());
+    workUnit.setProp(STAGEABLE_TABLE_METADATA_KEY, HiveSource.GENERICS_AWARE_GSON.toJson(destinationTable));
+    if (!Strings.isNullOrEmpty(partitionName)) {
```

Comment:
This partition-related branch doesn't seems to be consistent with the way a partition parameter is handle in `HiveWorkUnit(HiveDataset hiveDataset, Partition partition)`, which not only set partition name but other properties. 

The essential question here is, are there use cases when a `HiveProcessingEntity` is representing partitions and they are requested to be materialized ? If so, the construction of `HiveWorkUnit` should be modified.

Reply:
It's an interesting comment. My thought is that this is a bit easier to use, you can do
`tableCopyWorkUnit(dataset, destinationTable, "part=part123")`
or
`tableCopyWorkUnit(dataset, destinationTable, partition.getName())`

The first one does not require doing a lookup in the Hive metastore, while if you have the partition object you can always use the second method.

The main motivation for doing it this way, though, is that it was done this way before the refactoring. I have no problem refactoring even more to make users pass a partition if necessary.